### PR TITLE
DMs Styling Part 2 - Layout and Header

### PR DESCRIPTION
**Pull request by rickyrombo**
### Description

- Fixes the layout to match figma, especially with regards to the header
  - Note: "cog" and "compose" icon buttons not done
- Makes the layout responsive
- ProxyMemoizes getting the other user from a chat
- Creates a `ChatUser` component for reusing the same layout for a user in the header and the sidebar

image
image

### Dragons

*Is there anything the reviewer should be on the lookout for? Are there any dangerous changes?*

### How Has This Been Tested?

*Please describe the tests that you ran to verify your changes. Provide repro instructions & any configuration.*

### How will this change be monitored?

*For features that are critical or could fail silently please describe the monitoring/alerting being added.*

### Feature Flags ###

*Are all new features properly feature flagged? Describe added feature flags.*

### CODE REVIEW

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/mjp-dms-styling-2

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/mjp-dms-styling-2

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/mjp-dms-styling-2

File: packages/web/src/components/page/Page.js [hunk 1]
```diff
@@ -20,7 +20,7 @@ const HEADER_MARGIN_PX = 32
 const MIN_GUTTER_WIDTH = 20
 
 // Responsible for positioning the header
-const HeaderContainer = ({ header, containerRef }) => {
+const HeaderContainer = ({ header, containerRef, useSearch }) => {
```
Comment (sddioulde) [inline packages/web/src/components/page/Page.js:23]: very nit: useSearch sounds like a hook, but it doesn't look like it is? maybe `shouldUseSearch`

**Review by sddioulde — APPROVED** [review]
nice. looks good.

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/mjp-dms-styling-2

**Review by piazzatron — APPROVED** [review]
Not sure if fixed on subsequent PRs, but found a bug where clicking back and forth between chats will append the same messages to the chatlist multiple times.

Otherwise looks GREAT

**Comment by piazzatron** [comment]
Also, same deal with navigating back to to `/messages` - we append chats multiple times

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/mjp-dms-styling-2

**Pull request merged**